src/bv2int.cpp

```diff
@@ -312,29 +312,9 @@ Term BV2Int::handle_bw_op(const Term & t,
     y = tmp;
   }
 
-  Term res;
-  if (op.prim_op == BVAnd) {
-    TermVec args = { fbv_and(), bv_width_term, x, y };
-    res = solver_->make_term(Apply, args);
-  } else if (op.prim_op == BVOr) {
-    TermVec args = { fbv_or(), bv_width_term, x, y };
-    res = solver_->make_term(Apply, args);
-  } else if (op.prim_op == BVXor) {
-    TermVec args = { fbv_xor(), bv_width_term, x, y };
-    res = solver_->make_term(Apply, args);
-  } else {
-    assert(false);
-  }
-
+  Term res = get_explicit_bw(op, bv_width, TermVec({x, y}));
   fterms_.push_back(res);
 
-  if (!lazy_bw_) {
```

Comment:
Where did this go? If we are in lazy mode then we do not want to push anything to `extra_assertions_`, as whatever we would push here would be pushed lazily later.

Reply:
Ahh I see, good catch. How do you feel about this fix? f0ea473e04b429456ed629d537bf1abf9db45bc4

I noticed that `gen_explicit_bw` was only used once so I got rid of it. Then, in the lazy case it passes a dummy vector and ignores the side effects, otherwise it passes `extra_assertions_` as usual.